src/test/java/com/exasol/bucketfs/ReadEnabledBucketTest.java

```diff
@@ -72,10 +74,32 @@ private ReadOnlyBucket createBucket() {
     }
 
     @Test
-    void testRequestListingSucceeds() throws BucketAccessException, IOException, InterruptedException {
-        simulateResponse("file", 200);
+    // [utest->dsn~bucket-lists-files-with-common-prefix~1]
+    void listRoot() throws Exception {
+        simulateResponse(lines("folder/b1.txt", "b.txt", "folder/a1.txt", "a.txt"), 200);
+        final List<String> actual = createBucket().listContents();
+        assertThat(actual, equalTo(List.of("a.txt", "b.txt", "folder/")));
+    }
+
+    @Test
+    // [utest->dsn~bucket-lists-files-with-common-prefix~1]
+    void listSubFolder() throws Exception {
+        simulateResponse(lines("folder/b1.txt", "b.txt", "folder/a1.txt", "a.txt"), 200);
+        final List<String> actual = createBucket().listContents("folder/");
+        assertThat(actual, equalTo(List.of("a1.txt", "b1.txt")));
+    }
+
+    @Test
```

Comment:
A test for the `a` + `a/b.txt` case seems to be missing.

Reply:
Just below 😃

Reply:
Damn. :-)